### PR TITLE
Make rule component and `hr` element consistent in styling

### LINKED ISSUE #5096: Remove distinction between horizontal rules with and without margin
## Component/pattern to amend

Depending on what comes after a line, we use no margin bottom (when followed by text), or .5rem margin-bottom (when followed by images, buttons, etc). This makes things look nicer, but complicates usage - pretty much every pr I review I leave a bunch of comments to apply the p-rule class to remove margins where they are not needed.

In the interest of simplifying the framework, let's drop this distinction and deprecate the p-rule class. This way, every hr will have a .5rem margin-bottom, no exxceptions to document or explain.

Preview:

image

### PR DESCRIPTION
**Pull request by petesfrench**
**This work comes from a conversation with Lyubo, he explained that we are deprecating the use of 'p-rule'. To avoid confusion for other devs I have updated the docs.**

## Done

- Deprecates the rule component
- Adds `.is-highlighted` and `.is-highlighted.is-accent` to the base rule
- Adds documentation and example for `hr.u-no-margin--bottom`, which will replace `.p-rule`
- Drive-bys:
  - Deprecates the bordered strip component
  - Some minor typo tweaks in documentation

Fixes #5096 , [WD-14224](https://warthogs.atlassian.net/browse/WD-14224)

## QA

- Open [demo](https://vanilla-framework-5241.demos.haus/docs/patterns/rule)
- Review updated documentation:
  - https://vanilla-framework-5241.demos.haus/docs/patterns/rule

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots

[if relevant, include a screenshot or screen capture]

[WD-14224]: https://warthogs.atlassian.net/browse/WD-14224?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/docs/examples/brochure/_25-25-50.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <section class="p-section">
   <div class="row">
-    <hr class="p-rule">
```
Comment (bartaz) [inline templates/docs/examples/brochure/_25-25-50.html:3]: Let's not remove these. With the update of CSS they work the same with or without class name, but it's safer to keep the class name (for future compatibility).

File: templates/docs/examples/patterns/rule/muted.html [hunk 2]
```diff
@@ -8,20 +8,20 @@
...
         </div>
         <div class="col-3">
-          <h2><a href="#">OpenStack&nbsp;›</a></h2>
-          <hr class="p-rule--muted u-hide--medium">
-          <h2><a href="#">Ceph&nbsp;›</a></h2>
-          <hr class="p-rule--muted u-hide--medium">
-          <h2><a href="#">LXD&nbsp;›</a></h2>
-          <hr class="p-rule--muted u-hide--medium">
-          <h2><a href="#">MAAS&nbsp;›</a></h2>
+          <h2><a href="#">OpenStack&nbsp;&rsaquo;</a></h2>
```
Comment (petesfrench) [inline templates/docs/examples/patterns/rule/muted.html:18]: @bartaz Is this an appropriate change?

Comment (bartaz) [reply templates/docs/examples/patterns/rule/muted.html:18]: Yes, I guess it's safer this way. If that's how you usually encode these in content, lets keep it. Thanks.

File: templates/docs/patterns/rule.md [hunk 3]
```diff
@@ -12,15 +12,15 @@ It indicates the direction in which we want the viewer to read the content.
...
+Main page sections are often separated by horizontal rules. You can use the `p-rule` class name to create a default horizontal rule. Note: Currently, the class does not provide any visual changes. 
 
 <div class="embedded-example"><a href="/docs/examples/patterns/rule/default" class="js-example">
 View example of a default horizontal rule
 </a></div>
 
 ## Muted
 
-Muted rules are usually used within sections of content. You can add a muted rule to make it less prominent by using the `p-rule--muted` class name. This is useful for separating smaller blocks of content without drawing too much attention to the rule itself.
+Muted rules are usually used within sections of content. You can add a muted rule to make it less prominent by using the `is-muted` class name. This is useful for separating smaller blocks of content without drawing too much attention to the rule itself.
```
Comment (bartaz) [inline templates/docs/patterns/rule.md:23]: We are keeping the rule component, so worth documenting this as `p-rule--muted` still. Not a flag on `hr` element.
```suggestion
Muted rules are usually used within sections of content. You can add a muted rule to make it less prominent by using the `p-rule--muted` class name. This is useful for separating smaller blocks of content without drawing too much attention to the rule itself.
```

File: templates/docs/examples/patterns/rule/muted.html [hunk 4]
```diff
@@ -8,20 +8,20 @@
 <div class="p-section">
   <div class="row--25-75">
     <div class="col">
-      <hr class="p-rule">
+      <hr>
```
Comment (bartaz) [inline templates/docs/examples/patterns/rule/muted.html:11]: This is example of rule component, please keep the `p-rule` and `p-rule--muted` class names in it.

```suggestion
      <hr class="p-rule">
```

File: templates/docs/examples/patterns/rule/muted.html [hunk 5]
```diff
@@ -8,20 +8,20 @@
...
         <div class="col-3">
-          <h2><a href="#">OpenStack&nbsp;›</a></h2>
-          <hr class="p-rule--muted u-hide--medium">
-          <h2><a href="#">Ceph&nbsp;›</a></h2>
-          <hr class="p-rule--muted u-hide--medium">
-          <h2><a href="#">LXD&nbsp;›</a></h2>
-          <hr class="p-rule--muted u-hide--medium">
-          <h2><a href="#">MAAS&nbsp;›</a></h2>
+          <h2><a href="#">OpenStack&nbsp;&rsaquo;</a></h2>
+          <hr class="is-muted u-hide--medium">
```
Comment (bartaz) [inline templates/docs/examples/patterns/rule/muted.html:19]: Here and in following lines:
```suggestion
          <hr class="p-rule--muted u-hide--medium">
```

File: templates/docs/examples/brochure/_50-50.html [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <section class="p-section">
   <div class="p-section--shallow">
     <div class="row">
-      <hr class="p-rule">
+      <hr>
```
Comment (bartaz) [inline templates/docs/examples/brochure/_50-50.html:4]: As this is showcased as the rule component example, please bring `p-rule` class names back in this file.

File: templates/docs/examples/brochure/_25-75-nested.html [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <section class="p-section">
   <div class="row">
-    <hr class="p-rule">
```
Comment (bartaz) [inline templates/docs/examples/brochure/_25-75-nested.html:3]: As this is showcased as the rule component example, please bring `p-rule` class names back in this file.

Comment (jmuzina) [reply templates/docs/examples/brochure/_25-75-nested.html:3]: I think we need this `.p-rule` class reintroduced into this template, as Bartek suggested?

File: templates/docs/examples/brochure/_50-50.html (hunk 6 above)
Comment (jmuzina) [reply templates/docs/examples/brochure/_50-50.html:4]: I think the `p-rule` class needs to be re-added here

File: templates/docs/examples/patterns/rule/muted.html (hunk 4 above)
Comment (jmuzina) [reply templates/docs/examples/patterns/rule/muted.html:11]: I think we still need these rule classes re-added

File: templates/docs/patterns/rule.md (hunk 3 above)
Comment (jmuzina) [reply templates/docs/patterns/rule.md:23]: Should this still be `p-rule--muted`?

File: templates/docs/examples/patterns/rule/muted.html (hunk 5 above)
Comment (jmuzina) [reply templates/docs/examples/patterns/rule/muted.html:19]: Should this also be `p-rule--muted` as Bartek suggested?

**Review by jmuzina — COMMENTED** [review]
I think some of the points Bartek raised in his reviews still need to be addressed, unless there was a change I'm not aware of

**Review by jmuzina — CHANGES_REQUESTED** [review]
I believe this should be set to version `4.15.1` as `4.15.0` has been released

**Comment by petesfrench** [comment]
@jmuzina Apologies, I think something went wrong during a rebase and some work was lost.

**Comment by jmuzina** [comment]
This branch was created before #5300 and doesn't have its changes in it, causing some unexpected Percy results. We should merge `main` into this PR to fix that

**Comment by jmuzina** [comment]
@pastelcyborg caught a couple of visual quirks that I missed, we've left comments in this Percy build:

https://percy.io/bb49709b/vanilla-framework/builds/35904196/changed/1963530132?browser=chrome&browser_ids=59&group_snapshots_by=similar_diff&subcategories=approved&viewLayout=overlay&viewMode=new&width=1280&widths=375%2C1280

The examples that use colors bars seem like there might be unintended space between the bars and the rule above them, so let's wait until @lyubomir-popov is back on Monday to determine if this is expected or not.

Screenshot 2024-08-16 at 10 05 25 AM

**Comment by lyubomir-popov** [comment]
to keep things logical, mmy proposal is: 
- make the u-no-margin bottom class work correctly with hrs (factoring in the rule height), and change the exampel to use that
- deprecate rather than outright remove the p-rule class, to avoid breaking changes

**Comment by jmuzina** [comment]
>     * make the u-no-margin bottom class work correctly with hrs (factoring in the rule height, so elements after don't drift from the baseline grid), and change the stats example above to use that
I took a look and it looks like we already have the SCSS for this (`hr.u-no-margin--bottom`) ([src](https://github.com/canonical/vanilla-framework/blob/154e8584179e4d6fca56745e3c7d03fbb9526db1/scss/_base_hr.scss#L38-L41)) which offsets margin by `-1px` to account for HR height ([src](https://github.com/canonical/vanilla-framework/blob/154e8584179e4d6fca56745e3c7d03fbb9526db1/scss/_base_placeholders.scss#L206-L212))


>     * deprecate rather than outright remove the p-rule class, to avoid breaking changes
Agree, this is the safest approach

**Comment by jmuzina** [comment]
@lyubomir-popov I've gone ahead and updated this to deprecate the no-margin-bottom removal rather than remove it, and change the We're Canonical examples to use `hr.u-no-margin--bottom` which works as expected.

**Comment by lyubomir-popov** [comment]
Should we menton it is deprecated and whats the right way to do it (hr)

**Comment by lyubomir-popov** [comment]
And can I please ask for a driveby - mention somehwere on the hr component that all hrs other than the one above an h2 should be muted.

**Comment by lyubomir-popov** [comment]
from our conversation  - is-muted is-highlighted is-fixed-width u-no-margin--bottom classes need to be documented, 2 of them already are

File: releases.yml [hunk 8]
```diff
@@ -8,6 +8,14 @@
       url: /docs/base/forms#fieldset
       status: New
       notes: We've introduced <code>is-required</code> to legend fieldset elements.
+    - component: Rule
+      url: /docs/patterns/rule#default
+      status: Deprecated
+      notes: We've deprecated the removal of bottom margin on the <code>.p-rule</code>, <code>.p-rule--highlight</code>, and <code>.p-rule--muted</code> classes. Use the <a href="/docs/base/separators#no-bottom-margin">HR with no bottom margin</a> instead.
```
[the remark this answers is not in the document]
Comment (jmuzina) [reply releases.yml]: Revised

**Review by jmuzina — APPROVED** [review]

**Pull request merged**